src/main/java/org/jboss/dmr/ValueExpressionResolver.java

```diff
@@ -194,14 +195,40 @@ protected String resolvePart(String name) {
         } else if (":".equals(name)) {
             return File.pathSeparator;
         }
-        // First check for system property, then env variable
+        // First check for system property, then env variables
         String val = System.getProperty(name);
+
+        if (val == null) {
+            // See if an env var is defined
+            String envVar = replaceNonAlphanumericByUnderscores(name);
+            envVar = envVar.toUpperCase(Locale.ENGLISH);
+            envVar = envVar.replace(".", "_");
+            val = System.getenv(envVar);
+        }
+
         if (val == null && name.startsWith("env."))
             val = System.getenv(name.substring(4));
 
         return val;
     }
 
+
+    private String replaceNonAlphanumericByUnderscores(final String name) {
+        int length = name.length();
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < length; i++) {
+            char c = name.charAt(i);
```

Comment:
```suggestion
        int cp;
        for (int i = 0; i < length; i += Character.charCount(cp)) {
            cp = Character.toUpperCase(name.codePointAt(i));
```
This is from memory without the crutch of an IDE but I think this will simplify things a little as well as behaving correctly for Unicode things.